### PR TITLE
Replace descriptions on tables with overflow text and tooltip #445

### LINKED ISSUE #445: Replace description icon in tables with actual description but with tooltip?
Likely need datagateway-common's arrow tooltip for detecting the size.

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description

Changed current tooltip to be overflow text of the description, rather than tooltip icon

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking

closes #445

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
Can you add unit tests for the overflow function in utils please

File: src/utils.tsx [hunk 1]
```diff
@@ -88,6 +90,42 @@ export const formatDateTimeStrings = (
   return formattedDate;
 };
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const OverflowTip = ({ children }: any) => {
+  const [isOverflowed, setIsOverflow] = React.useState(false);
+  const overflowElementRef = useRef<HTMLInputElement | null>(null);
+  React.useEffect(() => {
```
Comment (joelvdavies) [inline src/utils.tsx:97]: If you use the column options to reset the column size it doesn't update this e.g.
tooltipBug

File: src/catalogue/items/catalogueItemsTable.component.test.tsx [hunk 2]
```diff
@@ -216,20 +208,21 @@ describe('Catalogue Items Table', () => {
...
 
     await waitFor(() => {
       expect(
-        screen.queryByText(
+        screen.queryAllByText(
           'Precision energy meters for accurate measurements. 26'
-        )
-      ).not.toBeInTheDocument();
+        ).length
+      ).toBe(1);
```
Comment (joelvdavies) [inline src/catalogue/items/catalogueItemsTable.component.test.tsx:225]: I don't think this part of the test makes sense without checking that the length is 2 earlier or something as it could also be that it only finds one regardless.

File: src/catalogue/items/catalogueItemsTable.component.test.tsx [hunk 3]
```diff
@@ -200,14 +200,6 @@ describe('Catalogue Items Table', () => {
 
     await ensureColumnsVisible(['Description']);
 
-    await waitFor(() => {
-      expect(
-        screen.getByLabelText(
-          'Catalogue item description: Precision energy meters for accurate measurements. 26'
-        )
-      ).toBeInTheDocument();
-    });
```
Comment (joelvdavies) [inline src/catalogue/items/catalogueItemsTable.component.test.tsx:209]: Was this not needed to prevent test flakiness by ensuring it finishes loading?

**Review by joelvdavies — COMMENTED** [review]
Works well for me, just two minor test comments.

File: src/catalogue/items/catalogueItemsTable.component.test.tsx (hunk 3 above)
Comment (MatteoGuarnaccia5) [reply src/catalogue/items/catalogueItemsTable.component.test.tsx:209]: The test seems to work fine without it, can add it back in though to prevent flakiness

File: src/catalogue/items/catalogueItemsTable.component.test.tsx (hunk 2 above)
Comment (joelvdavies) [reply src/catalogue/items/catalogueItemsTable.component.test.tsx]: @MatteoGuarnaccia5 Sorry I meant it the test for it should be modified rather than removed. Its a good thing to check it hides again. (Also just occurred to me the `infoIcon` should be renamed now as well`)

**Review by joelvdavies — APPROVED** [review]
LGTM

**Review by joelvdavies — APPROVED** [review]

**Review by joshuadkitenge — APPROVED** [review]
